photon-ml/src/main/scala/com/linkedin/photon/ml/Driver.scala

```diff
@@ -517,7 +514,16 @@ object Driver {
     val params = PhotonMLCmdLineParser.parseFromCommandLine(args)
 
     /* Configure the Spark application and initialize SparkContext, which is the entry point of a Spark application */
-    val sc: SparkContext = SparkContextConfiguration.asYarnClient(new SparkConf(), params.jobName, params.kryo)
+    val sc = SparkContextConfiguration.asYarnClient(new SparkConf(), params.jobName, params.kryo)
+    val configuration = sc.hadoopConfiguration
+
+    assert(!IOUtils.isDirExisting(params.outputDir, configuration),
+      s"Output directory ${params.outputDir} already exists!" )
```

Comment:
This feels more like a ```require``` to me. The difference between ```assert``` and ```require``` is subtle, but I'd characterize it like this:

**assert** - ensure that we are in a sane state as it relates to internal code consistency (i.e. in cases where a failure might be the code's fault)

**require** - ensure that we can make certain assumptions about user-supplied information (i.e. in cases where a failure might be the user's fault)

The other distinction is that ```assert``` is elidable, meaning that assertions can be disabled during build.